### PR TITLE
✨ Investigator resource

### LINKED ISSUE #165: Investigator endpoint
Write an /investigator endpoint that supports basic CRUD operations:

- [ ] GET /investigators - return investigators from dataservice
- [ ]  GET /investigators/{kf_id} - Get an investigator by id
- [ ]  POST /investigators - Creates a investigator in the dataservice
- [ ]  PATCH /investigators/{kf_id} - Partially updates an investigator by id
- [ ]  DELETE /investigators/{kf_id} - Deletes an investigator

### PR DESCRIPTION
**Pull request by dankolbman**
Adds endpoints for investigator.

resolves #165

### CODE REVIEW

File: dataservice/api/investigator/resources.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+              Investigator
+        """
+        try:
+            investigator = Investigator.query.filter_by(kf_id=kf_id).one()
+        except NoResultFound:
+            abort(404, 'could not find {} `{}`'
+                  .format('Investigator', kf_id))
+        return InvestigatorSchema().jsonify(investigator)
+
+    def patch(self, kf_id):
```
Comment (znatty22) [inline dataservice/api/investigator/resources.py]: See comments on #175, comments for this PR are going to be very similar

File: tests/investigator/test_investigator_resources.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                                   headers=self._api_headers())
+        resp = json.loads(response.data.decode('utf-8'))
+        self.assertEqual(response.status_code, 200)
+        self._test_response_content(resp, 200)
+
+        investigator = resp['results']
+        p = Investigator.query.first()
+        self.assertEqual(kf_id, investigator['kf_id'])
+
+    def test_patch_investigator(self):
```
Comment (znatty22) [inline tests/investigator/test_investigator_resources.py:53]: See comments on #175. Same comment here

File: dataservice/api/investigator/resources.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
+from flask import abort, request
+from sqlalchemy.orm.exc import NoResultFound
```
Comment (parimalak) [inline dataservice/api/investigator/resources.py]: NoResultFound seems to be not used anywhere

File: dataservice/api/investigator/resources.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        """
+        investigator = Investigator.query.get(kf_id)
+        if investigator is None:
+            abort(404, 'could not find {} `{}`'
+                  .format('investigator', kf_id))
+        return InvestigatorSchema().jsonify(investigator)
+
+    def patch(self, kf_id):
+        """
+        Update fields on an investigator and preserves other, non-specified
```
Comment (znatty22) [inline dataservice/api/investigator/resources.py]: Same comment here as in #175

File: tests/investigator/test_investigator_resources.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        '''
+        body = {
+            'name': name
+        }
+        response = self.client.post(url_for(INVESTIGATOR_LIST_URL),
+                                    headers=self._api_headers(),
+                                    data=json.dumps(body))
+        return response
+
+    def _test_response_content(self, resp, status_code):
```
Comment (znatty22) [inline tests/investigator/test_investigator_resources.py]: Remove method and calls to it

File: tests/investigator/test_investigator_resources.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
+import json
+
+from flask import url_for
+
+from dataservice.extensions import db
```
Comment (znatty22) [inline tests/investigator/test_investigator_resources.py]: Remove unused imports: db, Study

**Review by znatty22 — APPROVED** [review]
👍

**Review by parimalak — APPROVED** [review]

**Pull request merged**